### ISSUE TITLE
Teams manifest requirements aren't listed

### ISSUE THREAD

**Issue by iMicknl** [MicrosoftDocs/bot-docs#1294]
Almost all of the questions / issues regarding OAuth and Teams I receive are because of a faulty manifest. The manifest is missing `token.botframework.com` as an entry in the `validDomains` section, which is required by the Teams channel. If this is missing, the user won't see an error, pressing the Sign In card button will do nothing at all. 

This requirement is listed in the [v3 documentation of the Teams platform](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/authentication/auth-oauth-card#getting-started-with-oauthcard-in-teams), however many people miss it, since they are redirected to the v4 documentation...

Could this be added to the documentation somewhere? Or at least to the FAQ.

[StackOverflow example](https://stackoverflow.com/questions/57278351/oauth-card-in-ms-teams/57280507#57280507). [Github example](https://github.com/microsoft/BotBuilder-Samples/issues/1553#issuecomment-499279211) (But there are many more on SO and Github. It was also one of the most asked questions about Bot Framework during the company wide hackathon last week).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a9a43e68-83a3-1adc-0c55-e277ddb98c6b
* Version Independent ID: 5c46848d-bc20-480e-4571-00a283d930a2
* Content: [Add authentication to your bot via Azure Bot Service - Bot Service](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=aadv1%2Ccsharp%2Cbot-oauth#feedback)
* Content Source: [articles/v4sdk/bot-builder-authentication.md](https://github.com/MicrosoftDocs/bot-docs/blob/live/articles/v4sdk/bot-builder-authentication.md)
* Service: **bot-service**
* Sub-service: **abs**
* GitHub Login: @JonathanFingold
* Microsoft Alias: **v-jofing**

**Comment by emgrol** [comment]
Hi Mick, I'm taking a look at this now to see how we can get this content somewhere users will see it (either in the v4 docs or FAQ).

**Comment by iMicknl** [comment]
Thanks @emgrol! It would be great if the [following note](https://docs.microsoft.com/en-us/microsoftteams/add-bots) could be integrated or linked somewhere as well. 

>Adding a bot by GUID, for anything other than testing purposes, is not recommended. Doing so severely limits the functionality of a bot. Bots in production use should be added to Teams as part of an app. See Create a bot and Test and debug your Microsoft Teams bot

**Comment by emgrol** [comment]
Hi @iMicknl, @JonathanFingold and I discussed the content and decided to open issue [#1314](https://github.com/MicrosoftDocs/bot-docs/issues/1314) to cover portions related to 1) the token.botframework.com as an entry in the validDomains section and 2) how your note in the previous comment about adding by GUID can be worked in.

In the mean time we decided to add the code snippet in the comment [here](https://github.com/microsoft/BotBuilder-Samples/issues/1553) from @mdrichardson to the **Prepare your bot code** section of the [current v4 authentication article.](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=aadv1%2Ccsharp%2Cbot-oauth) I've reached out to @mdrichardson about getting a Javascript sample too.

**Comment by mdrichardson** [comment]
@emgrol Now that I've looked into this a little more, I'd recommend **not** adding any of that code into the docs. The SDK should handle all of that now. Should be [as simple as this C# sample shows](https://github.com/microsoft/BotBuilder-Samples/blob/master/samples/csharp_dotnetcore/46.teams-auth/Bots/TeamsBot.cs#L34) ([Node](https://github.com/microsoft/BotBuilder-Samples/blob/master/samples/javascript_nodejs/46.teams-auth/bots/teamsBot.js#L27)).

**Comment by emgrol** [comment]
We've updated the [authentication article](https://docs.microsoft.com/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=aadv1%2Ccsharp%2Cbot-oauth#feedback) to have a Teams Authentication section. It now references the teams-auth samples and shows users which code snippets to add along with a description of the TeamsActivityHandler. 

I've filed a new [issue](https://github.com/MicrosoftDocs/bot-docs/issues/1314) for more guidance on the content regarding the " missing `token.botframework.com` as an entry in the `validDomains` section".
